pallets/order-book/src/lib.rs

```diff
@@ -0,0 +1,439 @@
+            + MaybeDisplay
+            + AtLeast32BitUnsigned
+            + Copy
+            + Ord
+            + PartialEq
+            + Eq
+            + MaxEncodedLen
+            + scale_info::TypeInfo;
+        type MaxOpenedLimitOrdersForAllOrderBooksPerUser: Get<u32>;
+        type MaxLimitOrdersForPrice: Get<u32>;
+        type WeightInfo: WeightInfo;
+    }
+
+    pub type OrderBookId<T> = TradingPair<<T as assets::Config>::AssetId>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn order_books)]
+    pub type OrderBooks<T: Config> =
+        StorageMap<_, Blake2_128Concat, OrderBookId<T>, OrderBook<T>, OptionQuery>;
+
+    #[pallet::storage]
+    #[pallet::getter(fn limit_orders)]
+    pub type LimitOrders<T: Config> = StorageDoubleMap<
+        _,
+        Blake2_128Concat,
+        OrderBookId<T>,
+        Blake2_128Concat,
+        T::OrderId,
+        LimitOrder<T>,
+        OptionQuery,
+    >;
+
+    #[pallet::storage]
+    #[pallet::getter(fn prices)]
+    pub type Prices<T: Config> = StorageDoubleMap<
+        _,
+        Blake2_128Concat,
+        OrderBookId<T>,
+        Blake2_128Concat,
+        T::Balance,
+        BoundedVec<T::OrderId, T::MaxLimitOrdersForPrice>,
+        OptionQuery,
+    >;
+
+    #[pallet::storage]
+    #[pallet::getter(fn user_limit_orders)]
+    pub type UserLimitOrders<T: Config> = StorageDoubleMap<
+        _,
+        Blake2_128Concat,
+        T::AccountId,
+        Blake2_128Concat,
+        OrderBookId<T>,
+        BoundedVec<T::OrderId, T::MaxOpenedLimitOrdersForAllOrderBooksPerUser>,
+        OptionQuery,
+    >;
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        OrderPlaced {
+            order_book_id: OrderBookId<T>,
+            order_id: T::OrderId,
+            owner_id: T::AccountId,
+        },
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        /// Trading pair currently reached its capacity
+        OrderLimitReached,
+        /// Price in given order exceeds allowed limits for the trading pair
+        PriceExceedsLimits,
+        /// Lifespan exceeds defined limits
+        InvalidLifespan,
+        /// Order book does not exist for this trading pair
+        UnknownOrderBook,
+        /// Order book already exists for this trading pair
+        OrderBookAlreadyExists,
+        /// Cannot insert the limit order
+        InsertLimitOrderError,
+        /// Cannot delete the limit order
+        DeleteLimitOrderError,
+        /// There is not enough liquidity in the order book to cover the deal
+        NotEnoughLiquidity,
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::call_index(0)]
+        #[pallet::weight(<T as Config>::WeightInfo::create_orderbook())]
+        pub fn create_orderbook(
+            origin: OriginFor<T>,
+            _dex_id: T::DEXId,
+            order_book_id: OrderBookId<T>,
+            _tick_size: T::Balance,
```

Comment:
It seems that this type is not used anywhere else in this repo. Should it be just `Balance` instead?

Reply:
Types will be changed with my next commit

Reply:
And no, we should use T::Balance instead of Balance in all pallets

Reply:
Then [this type](https://github.com/sora-xor/sora2-network/pull/388/files#diff-2caeaf0a4f9f693d5b36b62bcf857021fa00942c659d2a90f126238bf6c99270R290) should be changed to T::Balance (just in case it's missed)